lib/pf/firewallsso/Checkpoint.pm

```diff
@@ -0,0 +1,110 @@
+package pf::firewallsso::Checkpoint;
+
+=head1 NAME
+
+pf::firewallsso::Checkpoint
+
+=head1 SYNOPSIS
+
+The pf::firewallsso::Checkpoint module implements an object oriented interface
+to update the Checkpoint user table.
+
+=cut
+
+
+use strict;
+use warnings;
+
+use base ('pf::firewallsso');
+
+use POSIX;
+use Log::Log4perl;
+
+use pf::config;
+sub description { 'Checkpoint Firewall' }
+use pf::util::radius qw(perform_rsso);
+use pf::node qw(node_view);
+use pf::accounting qw(node_accounting_current_sessionid);
+
+=head1 METHODS
+
+=head2 action
+
+Perform a radius accounting request based on the registered status of the node and his role.
+
+=cut
+
+sub action {
+    my ($self,$firewall_conf,$method,$mac,$ip,$timeout) = @_;
+    my $logger = Log::Log4perl::get_logger(ref($self));
+
+    my $node_info = node_view($mac);
+
+    my @categories = split(/,/, $ConfigFirewallSSO{$firewall_conf}->{categories});
+    if (
+        defined($node_info) &&
+        (ref($node_info) eq 'HASH') &&
+        $node_info->{'status'} eq $pf::node::STATUS_REGISTERED &&
+        (grep $_ eq $node_info->{'category'}, @categories)
+    ){
+        my $username = $node_info->{'pid'};
+        $username = $node_info->{'last_dot1x_username'} if ( $ConfigFirewallSSO{$firewall_conf}->{'uid'} eq '802.1x');
+        return 0 if ( $ConfigFirewallSSO{$firewall_conf}->{'uid'} eq '802.1x' && $node_info->{'last_dot1x_username'} eq '');
```

Comment:
Should also include a logger statement from my point of view since we are exiting the method due to "unmet dependencies"...